src/components/include/utils/threads/thread.h

```diff
@@ -220,6 +235,34 @@ class Thread {
   sync_primitives::ConditionalVariable state_cond_;
 
  private:
+  void SetState(const ThreadState thread_state) {
+    thread_state_ = thread_state;
+  }
+
+  void SwitchOnState(const ThreadState thread_state) {
+    if (IsSwitchOnState(thread_state)) {
+      return;
+    }
+
+    thread_state_ ^= thread_state;
+  }
+
+  void SwitchOffState(const ThreadState thread_state) {
+    if (!IsSwitchOnState(thread_state)) {
+      return;
+    }
+
+    thread_state_ ^= thread_state;
+  }
+
+  bool IsSwitchOnState(const ThreadState thread_state) const {
+    return thread_state & thread_state_;
+  }
+
+  bool IsNoneState() const {
+    return kThreadStateNone == thread_state_;
+  }
```

Comment:
Why do we need this if we already have `IsSwitchOnState`?

Reply:
@YarikMamykin 
IsSwitchOnState in this case will not work correctly 0 & 0

Reply:
@SNiukalov oh I see, because of that bit mask check...Maybe we could add specific if-statement for `None` case?